src/qibo/transpiler/unitary_decompositions.py

```diff
@@ -71,15 +71,14 @@ def calculate_psi(unitary, magic_basis=magic_basis, backend=None):
     )
     # construct and diagonalize UT_U
     ut_u = backend.np.transpose(u_magic, (1, 0)) @ u_magic
-    # When the matrix given to np.linalg.eig is a diagonal matrix up to machine precision the decomposition
-    # is not accurate anymore. decimals = 20 works for random 2q Clifford unitaries.
-    if backend.__class__.__name__ == "TensorflowBackend":
+    if backend.__class__.__name__ != "PyTorchBackend":
+        # eig seems to have a different behavior based on backend/hardware,
+        # use np.round to increase precision seems to fix the issue
```

Comment:
`round` is not being used anymore but it's still in the comments

Reply:
I had to put back the np.round to make tests pass

Reply:
This seems like a bigger underlying problem that should be solved in a different PR. Maybe an issue here should be opened about it.

Reply:
The problem was known before and we did not manage to find a more elegant solution.
Here the old discussion in qibolab:  https://github.com/qiboteam/qibolab/pull/501
If you want I can opens a new issue here in Qibo